app/assets/javascripts/channels/export.js

```diff
@@ -1,37 +1,42 @@
-function subscribeExportChannel(uuid, callback) {
-  App.download = App.cable.subscriptions.create(
+function subscribeExportChannel(uuid, startExportJobFtn, completionCallback) {
+  App.cable.subscriptions.create(
     { channel: "ExportChannel", uuid: uuid },
     {
       connected: function() {
-        callback();
+        startExportJobFtn();
       },
 
-      disconnected: function() {},
+      // Called when the WebSocket connection is closed, either by server or by client-side stale connection monitor.
+      // If the connection closes before we receive a response, count that as a failed export and unsubscribe.
+      // Reconnecting is not useful because we can't tell whether the response was sent while we were disconnected.
+      disconnected: function() {
+        const errorText = "We're sorry, your download has failed. That happens sometimes for response sets over several MB's in size.\n\n" +
```

Comment:
I'd like to add instrumentation here, so we can keep track of how often this failure happens. What's the best way to do that? New Relic PageAction event?

Reply:
Yeah, that's a better suggestion. Implemented.